src/odemis/gui/model/__init__.py

```diff
@@ -1554,31 +1568,86 @@ def moveStageToView(self):
     def moveStageTo(self, pos):
         """
         Request an absolute move of the stage to a given position
+
         pos (tuple of 2 float): X, Y absolute coordinates
         :return (None or Future): a future (that allows to know when the move is finished)
         """
         if not self._stage:
             return None
 
-        move = {"x": pos[0], "y": pos[1]}
+        move = self.clipToStageLimits({"x": pos[0], "y": pos[1]})
 
-        # clip to the range of the axes
-        for ax, p in move.items():
-            axis_def = self._stage.axes[ax]
-            if (hasattr(axis_def, "range") and
-                not axis_def.range[0] <= p <= axis_def.range[1]
-               ):
-                p = max(axis_def.range[0], min(p, axis_def.range[1]))
-                move[ax] = p
-                logging.info("Restricting stage axis %s move to %g mm due to stage limit",
-                             ax, p * 1e3)
-
-        logging.debug("Requesting stage to move to %s m", move)
+        logging.debug("Requesting stage to move to %s mm in x direction and %s mm in y direction",
+                      move["x"] * 1e3, move["y"] * 1e3)
         f = self._stage.moveAbs(move)
         self._fstage_move = f
         f.add_done_callback(self._on_stage_move_done)
         return f
 
+    def clipToStageLimits(self, pos):
+        """
+        Clip current a position in x/y direction to the maximum allowed stage limits.
+
+        :param pos (dict): Position to be clipped with keys "x" and "y"
+        :return(dict): Position clipped to the stage limits with keys "x" and "y"
+        """
+        stage_limits = self._getStageLimitsXY()
+        if not stage_limits["x"][0] <= pos["x"] <= stage_limits["x"][1]:
+            pos["x"] = max(stage_limits["x"][0], min(pos["x"], stage_limits["x"][1]))
+            logging.info("Movements of the stage in x limited to %s mm, restricting movement to %s mm.",
+                         stage_limits["x"], pos["x"] * 1e3)
```

Comment:
`stage_limits["x"]` must also be `* 1e3` as you're indicating "mm" in the message. (same thing for y). However, this is a list. So it's not very convenient. I expect that this type of message is mostly only for debugging or understanding unexpected behaviour of the software. So you don't have to make it too pretty. It'd fine to report the position in meters.